aws/resource_aws_rds_cluster_test.go

```diff
@@ -2758,43 +2763,79 @@ resource "aws_rds_cluster" "test" {
 `, n)
 }
 
-func testAccAWSClusterConfig_EngineVersion(rInt int, engine, engineVersion string) string {
+func testAccAWSClusterConfig_EngineVersion(upgrade bool, rInt int) string {
 	return fmt.Sprintf(`
+data "aws_rds_engine_version" "test" {
+  engine             = "aurora-postgresql"
+  preferred_versions = ["9.6.3", "9.6.6", "9.6.8"]
+}
+
+data "aws_rds_engine_version" "upgrade" {
+  engine             = data.aws_rds_engine_version.test.engine
+  preferred_versions = data.aws_rds_engine_version.test.valid_upgrade_targets
+}
+
+locals {
+  parameter_group_name = %[1]t ? data.aws_rds_engine_version.upgrade.parameter_group_family : data.aws_rds_engine_version.test.parameter_group_family
+  engine_version       = %[1]t ? data.aws_rds_engine_version.upgrade.version : data.aws_rds_engine_version.test.version
+}
```

Comment:
Nit: since this looks similar to the `locals` defined below, you could potentially move this to its own `func` and use `composeConfig`